source/lac/tridiagonal_matrix.cc

```diff
@@ -249,7 +249,7 @@ TridiagonalMatrix<number>::compute_eigenvalues()
        &one,
        static_cast<number *>(nullptr),
        &info);
-  Assert(info == 0, ExcInternalError());
+  AssertThrow(info == 0, ExcInternalError());
```

Comment:
By way of context (just working through this myself): The documentation page at https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-c/top/lapack-routines/lapack-least-squares-and-eigenvalue-problem/lapack-least-squares-eigenvalue-problem-driver/symmetric-eigenvalue-problems-lapack-driver/stev.html says that the last argument should actually be called `ldz` and is an input argument. On the other hand, https://www.smcm.iqfr.csic.es/docs/intel/mkl/mkl_manual/lse/functn_stev.htm provides for the Fortran interface, which I believe we must be using here, where the last argument is an output argument `info`. It should be zero if operations succeed, and nonzero if they don't -- so an internal error is appropriate.